components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaProducer.java

```diff
@@ -223,6 +231,12 @@ public boolean hasNext() {
                             }
                         }
 
+                        if (innerMmessage.getHeader(KafkaConstants.OVERRIDE_TIMESTAMP) != null) {
+                            if (innerMmessage.getHeader(KafkaConstants.OVERRIDE_TIMESTAMP) instanceof Long) {
+                                innerTimestamp = ((Long) overrideTimeStamp).longValue();
```

Comment:
innerMmessage.getHeader(KafkaConstants.OVERRIDE_TIMESTAMP)  will be lost

Reply:
why it will be lost?

Reply:
Probably I am getting it wrong but should the innerTimestamp be set with value of innerMmessage.getHeader(KafkaConstants.OVERRIDE_TIMESTAMP) instead of  overrideTimeStamp.

Reply:
yes @vgaur , the `innerTimestamp` should get `innerMmessage.removeHeader(KafkaConstants.OVERRIDE_TIMESTAMP)` since now we are dealing here with inner messages